### PR TITLE
 Fix: IT fails at times unless reindex has been run (#1174) 

### PR DESCRIPTION
**Pull request by jessebrennan**

### CODE REVIEW

File: src/azul/azulclient.py [hunk 1]
```diff
@@ -106,10 +107,13 @@ def test_notifications(self, test_name: str, test_uuid: str) -> Tuple[List[dict]
         effective_bundle_fqids = set()
         for bundle_fqid in real_bundle_fqids:
             new_bundle_uuid = str(uuid.uuid4())
-            _, _, version = bundle_fqid.partition('.')
-            effective_bundle_fqids.add(new_bundle_uuid + '.' + version)
+            # Using a new version helps ensure that any aggregation will choose this entity.
+            # Failing to do this before caused https://github.com/DataBiosphere/azul/issues/1174
+            new_bundle_version = datetime.utcnow().isoformat().replace(':', '') + 'Z'
```
Comment (hannes-ucsc) [inline src/azul/azulclient.py]: Can you consolidate with this:

https://github.com/DataBiosphere/azul/blob/develop/scripts/copy_bundles.py#L227

by moving that code into the `azul.dss` module and reusing it here. You may have to add that module to the suite in `test_doctest.py`.

Just double check that the code I link to actually generates a version representing the current time in UTC.

File: src/azul/azulclient.py [hunk 2]
```diff
@@ -106,10 +107,13 @@ def test_notifications(self, test_name: str, test_uuid: str) -> Tuple[List[dict]
         effective_bundle_fqids = set()
         for bundle_fqid in real_bundle_fqids:
             new_bundle_uuid = str(uuid.uuid4())
-            _, _, version = bundle_fqid.partition('.')
-            effective_bundle_fqids.add(new_bundle_uuid + '.' + version)
+            # Using a new version helps ensure that any aggregation will choose this entity.
```
Comment (hannes-ucsc) [inline src/azul/azulclient.py]:
```suggestion
            # Using a new version helps ensure that any aggregation will choose the contribution from the test bundle over that by any other bundle not in the test set.
```

File: src/azul/azulclient.py (hunk 1 above)
Comment (jessebrennan) [reply src/azul/azulclient.py]: I missed
 > by moving that code into the azul.dss module and reusing it here. You may have to add that module to the suite in test_doctest.py.

On my first pass. It was simpler and seemed appropriate to put it in a separate PR which I did: #1363.

File: scripts/copy_bundles.py [hunk 3]
```diff
@@ -212,21 +213,15 @@ def _copy_version(self, version: str):
...
                 return mode
 
-    version_format = '%Y-%m-%dT%H%M%S.%fZ'
-
-    @classmethod
-    def _new_version(cls):
-        return datetime.utcfromtimestamp(time.time()).strftime(cls.version_format)
-
     @classmethod
     def _validate_version(cls, version: str):
```
Comment (hannes-ucsc) [inline scripts/copy_bundles.py]: We should move this too.

File: scripts/copy_bundles.py [hunk 4]
```diff
@@ -26,6 +26,8 @@
 from azul.dss import (
     MiniDSS,
     shared_dss_credentials,
```
Comment (hannes-ucsc) [inline scripts/copy_bundles.py]: https://github.com/DataBiosphere/azul/pull/1366/files#diff-61633c57a57b908ccba54e0971f8fd0dR26

Symbols in azul.dss should not duplicate the `dss` string. The way this is implemented here and now, `new_version` is a bit general and lacks context. But if references as `azul.dss.new_version` it becomes clear.

File: scripts/copy_bundles.py [hunk 5]
```diff
@@ -197,7 +197,7 @@ def _copy_bundle(self, bundle_uuid, bundle_version, manifest, attempt=0):
                                         creator_uid=0,
                                         files=manifest['files'])
         except SwaggerAPIException as e:
-            if e.code == 400 and e.reason == 'file_missing' and attempt < 10:
+            if e.code == 401 and e.reason == 'file_missing' and attempt < 10:
```
Comment (hannes-ucsc) [inline scripts/copy_bundles.py]: Remove this change.

File: src/azul/transformer.py [hunk 6]
```diff
@@ -751,10 +751,16 @@ def _select_latest(self, contributions: Sequence[Contribution]) -> MutableMappin
...
                     for entity in entities:
                         entity_id = entity['document_id']  # FIXME: the key 'document_id' is HCA specific
-                        bundle_version, _ = collated_entities.get(entity_id, ('', None))
-                        if bundle_version < contribution.bundle_version:
-                            collated_entities[entity_id] = contribution.bundle_version, entity
+                        cur_bundle_uuid, cur_bundle_version, cur_entity = \
+                            collated_entities.get(entity_id, (None, '', None))
+                        if cur_entity is not None and entity.keys() != cur_entity.keys():
+                            symmetric_difference = set(entity.keys()).symmetric_difference(cur_entity)
+                            logger.warning('Shape of `%s` entities does not match between bundles %s, version %s '
```
Comment (jessebrennan) [inline src/azul/transformer.py]:
```suggestion
                            logger.warning('Document shape of `%s` entity does not match between bundles %s, version %s '
```

**Pull request merged**